README.md

```diff
@@ -84,30 +72,55 @@ download methods were (1) `dx cat`, and (2) `cat` from a dxfuse mount point.
+database, located on `/var/dxfuse/metadata.db`. It stores files and
+directories in tables, indexed to speed up common queries.
 
-**NOTE `dxfuse -limitedWrite` mode was primarly designed to support spark file output**
+Load on the DNAnexus API servers and the cloud object system is carefully controlled. Bulk calls
+are used to describe data objects, and the number of parallel IO requests is bounded.
+
+dxfuse operations can sometimes be slow, for example, if the server is
+slow to respond, or has been temporarily shut down (503 mode). This
+may cause the filesystem to lose its interactive feel. Running it on a
+cloud worker reduces network latency significantly, and is the way it
+is used in the product. Running on a local, non cloud machine, runs
+the risk of network choppiness.
+
+# Limited Write Mode
+
+`dxfuse -limitedWrite` mode is primarly designed to support spark file output over the `file:///` protocol.
+
+Creating and writing to files is allowed when dxfuse is mounted with the `-limitedWrite` flag.
+Writing to files is **append only**. Any non-sequential writes will return `ENOTSUP`. Seeking or reading operations are not permitted while a file is being written.
 
 ## Supported operations
 
-`-limitedWrite` mode also enables the following operations: rename (mv), unlink (rm), mkdir (see below), and rmdir (empty folders only). Rewriting existing files is not permitted, nor is truncating existing files. 
+`-limitedWrite` mode enables the following operations: rename (mv, see [below](#rename-behavior)), unlink (rm), mkdir (see [below](#mkdir-behavior)), and rmdir (empty folders only). Rewriting existing files is not permitted, nor is truncating existing files.
 
 ### mkdir behavior
 
-All `mkdir` operations via dxfuse are treated as `mkdir -p`. This is because dxfuse does not present the realtime state of the project. Folders can be created outside of dxfuse (or in another dxfuse process), and therefore not be visible to the current running dxfuse. A subsequent `mkdir` --> `project-xxxx/newFolder` would return a 422 error, because dxfuse did not know the directory already exists. This design is due to spark behavior where multiple worker nodes sometimes attempt to create the same output directory. 
+All `mkdir` operations via dxfuse are treated as `mkdir -p`. This is because dxfuse does not present the realtime state of the project. Folders can be created outside of dxfuse (or in another dxfuse process), and therefore not be visible to the current running dxfuse. A subsequent `mkdir` --> `project-xxxx/newFolder` would return a 422 error, because dxfuse did not know the directory already exists. This design is due to spark behavior where multiple worker nodes sometimes attempt to create the same output directory.
+
+### rename behavior
+
+Rename does not allow removing the target file or directory because DNAnexus API does not support this functionality.
+XXX moved here from limitations.  Still relevant and correct?
```

Comment:
This is relevant and correct, dxfuse does not support atomic rename if the target exists. 
e.g.
```
$ ls -lht MNT/posix/file*
-r--r--r-- 1 root root 6 Aug 20 21:23 file1
-r--r--r-- 1 root root 6 Aug 20 21:23 file
$ mv MNT/posix/file MNT/posix/file1
mv: cannot move 'file' to 'file1': Operation not permitted
```
Supported in 2 distinct operations
```
$ rm MNT/posix/file1
$ mv MNT/posix/file MNT/posix/file1
```

Reply:
updated. Thanks!